frontend/packages/webterminal-plugin/src/utils.ts

```diff
@@ -0,0 +1,143 @@
+import { K8sResourceKind } from '@console/internal/module/k8s';
+
+export const getUpdatedComponentTimeout = (
+  existingTerminalExecComponent,
+  timeout,
+  terminalExecResource,
+) => {
+  if (!existingTerminalExecComponent && timeout) {
+    return [
+      {
+        name: 'web-terminal-exec',
+        container: {
+          env: [{ name: 'WEB_TERMINAL_IDLE_TIMEOUT', value: timeout }],
+          image: '',
+        },
+      },
+    ];
+  }
+  const terminalExecComponent = terminalExecResource?.container?.env?.find(
+    (e) => e.name === 'WEB_TERMINAL_IDLE_TIMEOUT',
+  );
+  if (terminalExecResource && !terminalExecComponent?.value && timeout) {
+    let newEnvVariables;
+    if (terminalExecResource?.container?.env) {
+      newEnvVariables = [
+        ...terminalExecResource.container.env,
+        { name: 'WEB_TERMINAL_IDLE_TIMEOUT', value: timeout },
+      ];
+    } else {
+      newEnvVariables = [{ name: 'WEB_TERMINAL_IDLE_TIMEOUT', value: timeout }];
+    }
+    terminalExecResource.container = {
+      ...terminalExecResource.container,
+      env: newEnvVariables,
+    };
+    return existingTerminalExecComponent;
+  }
+  if (terminalExecResource) {
+    if (timeout && terminalExecComponent?.value) {
+      terminalExecComponent.value = timeout;
+    } else if (terminalExecComponent && !timeout) {
+      terminalExecResource.container.env = terminalExecResource.container.env.filter(
+        (e) => e.name !== 'WEB_TERMINAL_IDLE_TIMEOUT',
+      );
+    }
+    return existingTerminalExecComponent;
+  }
+  if (timeout) {
+    return [
+      ...existingTerminalExecComponent,
+      {
+        name: 'web-terminal-exec',
+        container: {
+          env: [{ name: 'WEB_TERMINAL_IDLE_TIMEOUT', value: timeout }],
+          image: '',
+        },
+      },
+    ];
+  }
+  return existingTerminalExecComponent;
+};
+
+export const updatedWebTerminalExec = (
+  webTerminalExec: K8sResourceKind,
+  timeoutCheckBox: boolean,
+  terminalExecResource: K8sResourceKind,
+  csTimeout: string,
```

Comment:
as discussed, terminalExecResource is not a K8sResourceKind. Would be great if you can create a type definition for the both resources we used here. Let me know if you have any question.